python_modules/dagster/dagster/check/__init__.py

```diff
@@ -850,6 +852,24 @@ def opt_str_param(obj: object, param_name: str, default: Optional[str] = None) -
     return default if obj is None else obj
 
 
+def path_param(obj: object, param_name: str) -> str:
+    try:
+        return fspath(obj)
+    except TypeError:
+        raise _param_type_mismatch_exception(obj, (str, PathLike), param_name)
+
+
+def opt_path_param(
```

Comment:
This should probably have `@overload`s like other `opt` methods-- if a `default` is passed, then we know it returns a `Union[str, os.PathLike]`, otherwise its `Optional`.

Reply:
Ok, I made a few overloads, but I'm not sure from the docs how you are testing typing. I tried `➜ mypy python_modules/dagster/dagster/` but it found blocking issues elsewhere.

<summary>mypy output</summary>
<details>

```zsh
➜ mypy python_modules/dagster/dagster/
python_modules/dagster/dagster/core/storage/runs/sqlite/alembic/versions/__init__.py: error: Duplicate module named "versions" (also at "python_modules/dagster/dagster/core/storage/event_log/sqlite/alembic/versions/__init__.py")
python_modules/dagster/dagster/core/storage/runs/sqlite/alembic/versions/__init__.py: note: Are you missing an __init__.py? Alternatively, consider using --exclude to avoid checking one of them.
Found 1 error in 1 file (errors prevented further checking)
```

</details>